packages/rich-text/src/test/to-dom.js

```diff
@@ -101,3 +101,32 @@ describe( 'applyValue', () => {
 		} );
 	} );
 } );
+
+describe( 'toDom-SVG', () => {
+	let body;
+	beforeAll( () => {
+		const svg = { type: 'svg' };
+		const use = { type: 'use', attributes: { 'xlink:href': '#logo' } };
+		body = toDom( {
+			value: {
+				start: 0,
+				end: 1,
+				formats: [ [ svg ] ],
+				replacements: [ use ],
+				text: '\ufffc',
+			},
+		} ).body;
+	} );
```

Comment:
Could you add a test case to `spec` in the helpers file instead? Although, I'm not sure at all about supporting SVGs like this. Maybe we could instead use `contentEditable` on any format that wants to use it, like we do for footnotes?

Reply:
I've added cases to `spec` in 376cc58. Still, testing the expected namespaces is essential to the proposed changes so other additions were required.

<details><summary>Diff of reverted to-dom changes so it can be applied to verify test failures without them.</summary>

```diff
diff --git a/packages/rich-text/src/to-dom.js b/packages/rich-text/src/to-dom.js
index 12dab33700..305eebaf3e 100644
--- a/packages/rich-text/src/to-dom.js
+++ b/packages/rich-text/src/to-dom.js
@@ -8,9 +8,6 @@ import { isRangeEqual } from './is-range-equal';
 
 /** @typedef {import('./types').RichTextValue} RichTextValue */
 
-const NS_SVG = 'http://www.w3.org/2000/svg';
-const NS_XLINK = 'http://www.w3.org/1999/xlink';
-
 /**
  * Creates a path as an array of indices from the given root node to the given
  * node.
@@ -71,26 +68,10 @@ function append( element, child ) {
 	const { type, attributes } = child;
 
 	if ( type ) {
-		let addAttribute;
-		const isSVGTag = type === 'svg';
-		const isSVGContext = element.namespaceURI === NS_SVG;
-		if ( isSVGTag || isSVGContext ) {
-			child = element.ownerDocument.createElementNS( NS_SVG, type );
-			// Disables editing at top-level SVG elements.
-			if ( isSVGTag && ! isSVGContext ) {
-				child.setAttribute( 'contentEditable', false );
-			}
-			addAttribute = ( key, value ) => {
-				if ( key === 'xlink:href' )
-					child.setAttributeNS( NS_XLINK, key, value );
-				else child.setAttribute( key, value );
-			};
-		} else {
-			child = element.ownerDocument.createElement( type );
-			addAttribute = child.setAttribute.bind( child );
-		}
+		child = element.ownerDocument.createElement( type );
+
 		for ( const key in attributes ) {
-			addAttribute( key, attributes[ key ] );
+			child.setAttribute( key, attributes[ key ] );
 		}
 	}
 
```
</details>